tests/compiler/test_list.py

```diff
@@ -431,3 +431,73 @@ def test_653_list_attribute_unset(snippetcompiler):
         "The object __config__::Test (instantiated at {dir}/main.cf:6) is not complete:"
         " attribute bla ({dir}/main.cf:3) requires 1 values but only 0 are set",
     )
+
+
+def test_1422_gradual_array(snippetcompiler):
+    snippetcompiler.setup_for_snippet(
+        """
+entity Cluster:
+end
+
+implement Cluster using agentConfig
+
+implementation agentConfig for Cluster:
+    a = std::AgentConfig(autostart=true, agentname="", uri="local:" )
+    a.provides = [self, self.provides, self.provides]
```

Comment:
Why do we have two times self.provides here? Is this done on purpose?

Reply:
done on purpose, to make it more brittle.

Reply:
Ok.